flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/sql/SqlJsonQueryFunction.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.functions.sql;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlFunction;
+import org.apache.calcite.sql.SqlFunctionCategory;
+import org.apache.calcite.sql.SqlJsonQueryEmptyOrErrorBehavior;
+import org.apache.calcite.sql.SqlJsonQueryWrapperBehavior;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlLiteral;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.sql.type.OperandTypes;
+import org.apache.calcite.sql.type.ReturnTypes;
+import org.apache.calcite.sql.type.SqlTypeFamily;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.sql.type.SqlTypeTransforms;
+
+/** This class has been copied from Calcite to replace the return type VARCHAR(2000) with STRING. */
```

Comment:
I wonder if you can just create a wrapper for `SqlFunction` to avoid copy-pasting `SqlJsonQueryFunction` and `SqlJsonArrayFunction`. If you check the code for `SqlFunction`, you see that the return type inference is used only for:

* `SqlReturnTypeInference getReturnTypeInference()` which is just a getter
* `RelDataType inferReturnType(SqlOperatorBinding opBinding)` which is a wrapper call for SqlReturnTypeInference.inferReturnType`.

WDYT?

Reply:
@slinkydeveloper , I agree with the point you mention above. I considered about the wrapper and I'm not sure whether creating a wrapper for `SqlFunction` has no problem. But I will try to create a wrapper for `SqlFunction`.

Reply:
Thank you! If we can fix it wrapping the class, it's a better solution. Copying classes should always be our last resort :smile: